agent/proxycfg/ingress_gateway.go

```diff
@@ -91,6 +91,9 @@ func (s *handlerIngressGateway) handleUpdate(ctx context.Context, u UpdateEvent,
 		if !ok {
 			return fmt.Errorf("invalid type for response: %T", u.Result)
 		}
```

Comment:
The fix is to guard on `resp.Entry` being nil. One thing this doesn't handle is if the config entry is deleted. If it's deleted, I think the expectation would be that everything is reset back to the baseline. This can be done in a different PR though and likely isn't worth it if the api gateway work is completed.